### PR TITLE
new Image management + fix crop image issue

### LINKED ISSUE #2694: Never ask whether to add the product when coming from history
Never ask whether to add the product when coming from history (or search results by the way)

image

This happens when there's no or little network. The app wants to be helpful. It would make sense if we had a local list of existing products like implemented in the offline PR

### LINKED ISSUE #2701: 3.1.6: no longer offering to crop or diplay photos after taking them
**Checklist**
- [x] Bug Report - The issue you are reporting might be there already. Please search on the [issues tracker](https://github.com/openfoodfacts/openfoodfacts-androidapp/issues) before creating or reporting an issue.
- [] Before reporting, please ensure (if you can) that you have tested it with the latest (dev) version of the software.
- [x] Don't forget to tag your issue, it makes it easier to discover.
**Summary:** 

After scanning an item that does not exist in the database, Summarize your issue in one sentence (what goes wrong, what did you expect to happen)

**Steps to reproduce:** 

- Scan an item that does not exist in the OFF db
- take a photo (product image, ingredients or nutrition
- OFF-android skips the option to crop/rotate image
- goes straight to uploading the image
- after the image is upload it is never displayed in the app

**Expected behavior:** 

- Scan an item that does not exist in the OFF db
- take a photo (product image, ingredients or nutrition)
- OFF-android gives the option to crop/rotate image 
- image is displayed in app
- image is upload

(This behavior is observed in OFF 3.1.2)

**Observed behavior:** 

After taking photo no option is given to crop and rotate. Image is not displayed after returning to OFF app

**Device and Android version:** 

Samsung Galaxy S9
Android 9
OpenCamera 1.46
OpenFoodFacts 3.1.6 (freshly installed)

### PR DESCRIPTION
**Pull request by deniger**
+ new way to manage product images -> just click on product image to edit it
+ Fix #2701 an option is available to crop new images
+ Fix #2694 if network issue, do not ask user to add a new product

image
image
image

### CODE REVIEW

**Comment by teolemon** [comment]
``` 
2019-06-05 22:18:51.430 27577-27577/openfoodfacts.github.scrachx.openfood E/AndroidRuntime: FATAL EXCEPTION: main
    Process: openfoodfacts.github.scrachx.openfood, PID: 27577
    java.lang.ClassCastException: java.lang.Integer cannot be cast to java.lang.String
        at openfoodfacts.github.scrachx.openfood.images.ImageTransformation.getImageRotation(ImageTransformation.java:250)
        at openfoodfacts.github.scrachx.openfood.images.ImageTransformation.getInitialServerTransformation(ImageTransformation.java:186)
        at openfoodfacts.github.scrachx.openfood.images.ImageTransformation.getScreenTransformation(ImageTransformation.java:118)
        at openfoodfacts.github.scrachx.openfood.views.FullScreenImage.onStartEditExistingImage(FullScreenImage.java:433)
        at openfoodfacts.github.scrachx.openfood.views.FullScreenImage_ViewBinding$1.doClick(FullScreenImage_ViewBinding.java:55)
        at butterknife.internal.DebouncingOnClickListener.onClick(DebouncingOnClickListener.java:22)
        at android.view.View.performClick(View.java:7251)
        at android.view.View.performClickInternal(View.java:7228)
        at android.view.View.access$3500(View.java:802)
        at android.view.View$PerformClick.run(View.java:27843)
        at android.os.Handler.handleCallback(Handler.java:883)
        at android.os.Handler.dispatchMessage(Handler.java:100)
        at android.os.Looper.loop(Looper.java:214)
        at android.app.ActivityThread.main(ActivityThread.java:7116)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:492)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:925)
2019-06-05 22:18:51.480 1048-1406/? E/InputDispatcher: channel 'f16461b openfoodfacts.github.scrachx.openfood/openfoodfacts.github.scrachx.openfood.views.product.ProductActivity (server)' ~ Channel is unrecoverably broken and will be disposed!
2019-06-05 22:18:51.501 1048-1155/? E/system_server: Invalid ID 0x00000000.
2019-06-05 22:18:51.664 28406-28406/? E/crachx.openfoo: Unknown bits set in runtime_flags: 0x8000
2019-06-05 22:18:51.744 30161-30161/? E/cutils-trace: Error opening trace file: No such file or directory (2)
2019-06-05 22:18:51.797 1708-25353/? E/EastworldError: null InputStream
```

**Comment by teolemon** [comment]
This one is already solved
``` 
2019-06-05 22:58:04.673 27104-28019/openfoodfacts.github.scrachx.openfood D/OkHttp: --> GET https://ssl-api.openfoodfacts.org/api/v0/product/8410063005683.json?fields=image_small_url,selected_images,images,languages_codes,vitamins_tags,minerals_tags,amino_acids_tags,other_nutritional_substances_tags,image_front_url,image_ingredients_url,image_nutrition_url,url,code,traces_tags,ingredients_that_may_be_from_palm_oil_tags,additives_tags,allergens_hierarchy,manufacturing_places,nutriments,ingredients_from_palm_oil_tags,brands_tags,traces,categories_tags,ingredients_text,product_name,generic_name,ingredients_from_or_that_may_be_from_palm_oil_n,serving_size,allergens_tags,allergens,origins,stores,nutrition_grade_fr,nutrient_levels,countries,countries_tags,brands,packaging,labels_tags,labels_hierarchy,cities_tags,quantity,ingredients_from_palm_oil_n,image_url,link,emb_codes_tags,states_tags,creator,created_t,last_modified_t,last_modified_by,editors_tags,nova_groups,lang,purchase_places,nutrition_data_per,no_nutrition_data,other,other_information,conservation_conditions,recycling_instructions_to_discard,recycling_instructions_to_recycle,warning,customer_service,environment_infocard,environment_impact_level_tags
2019-06-05 22:58:04.673 27104-28019/openfoodfacts.github.scrachx.openfood D/OkHttp: User-Agent: Official Android App 3.1.6 Search
2019-06-05 22:58:04.673 27104-28019/openfoodfacts.github.scrachx.openfood D/OkHttp: --> END GET
    
    --------- beginning of crash
2019-06-05 22:58:04.675 27104-27104/openfoodfacts.github.scrachx.openfood E/AndroidRuntime: FATAL EXCEPTION: main
    Process: openfoodfacts.github.scrachx.openfood, PID: 27104
    java.lang.NullPointerException: Attempt to invoke interface method 'int java.util.Collection.size()' on a null object reference
        at java.util.HashSet.<init>(HashSet.java:118)
        at openfoodfacts.github.scrachx.openfood.views.FullScreenImage.loadLanguage(FullScreenImage.java:168)
        at openfoodfacts.github.scrachx.openfood.views.FullScreenImage.onImageTypeChanged(FullScreenImage.java:476)
        at openfoodfacts.github.scrachx.openfood.views.FullScreenImage_ViewBinding$4.onItemSelected(FullScreenImage_ViewBinding.java:89)
        at android.widget.AdapterView.fireOnSelected(AdapterView.java:957)
        at android.widget.AdapterView.dispatchOnItemSelected(AdapterView.java:946)
        at android.widget.AdapterView.access$300(AdapterView.java:55)
        at android.widget.AdapterView$SelectionNotifier.run(AdapterView.java:910)
        at android.os.Handler.handleCallback(Handler.java:883)
        at android.os.Handler.dispatchMessage(Handler.java:100)
        at android.os.Looper.loop(Looper.java:214)
        at android.app.ActivityThread.main(ActivityThread.java:7319)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:492)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:934)
2019-06-05 22:58:04.681 1110-6238/? W/ActivityTaskManager:   Force finishing activity openfoodfacts.github.scrachx.openfood/.views.FullScreenImage
2019-06-05 22:58:04.682 1110-28379/? I/DropBoxManagerService: add tag=data_app_crash isTagEnabled=true flags=0x2
2019-06-05 22:58:04.690 1110-1140/? I/ActivityManager: Showing crash dial
```

**Comment by deniger** [comment]
another suggestion: it's possible to open the new view if the user clicks on the image in a product list and in the scan view. Is it ok ?

image
image

**Comment by teolemon** [comment]
Having a hard time with the new Android Q gestural system
image
https://twitter.com/den_sat/status/1126779492763099137

**Comment by deniger** [comment]
Thanks for the feedback. I'll test on android Q.
To fix this, we should use some method available in API 29 and so upgrade the compile sdk version:
https://github.com/openfoodfacts/openfoodfacts-androidapp/pull/2672

this PR changes it to 28 but we should use 29 then

**Comment by teolemon** [comment]
Step 1: 40015071111011 (https://barcode.tec-it.com/en/EAN13?data=40015071111011 )
Step 2: click on the front image
Step 3: select ES (because not selected by default)
Step 4: Edit image
Step 5: :boom:

File: app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -284,12 +284,96 @@
         <item>zu</item>
     </string-array>
 
+    <!--    Why using a different order than UI ?...-->
```
Comment (teolemon) [inline app/src/main/res/values/strings.xml]: might not be used anymore

File: app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -1026,8 +1124,12 @@
     <string name="add_a_list">Add a list</string>
     <string name="todo">TODO</string>
     <string name="edit_image">Edit Image</string>
+    <string name="unselect_image">Unselect Image</string>
+    <string name="choose_default_language">Select product language</string>
     <string name="barcode">Barcode</string>
     <string name="productAdditivesNone">No additives</string>
     <string name="error_uploading_photo">Something went wrong while uploading a new photo</string>
     <string name="history_network_error">The product couldn\'t be loaded. Please try again when you are back online.</string>
+    <string name="cant_modify_if_refreshing">Please, wait the end of the uploading task</string>
```
Comment (teolemon) [inline app/src/main/res/values/strings.xml]:
```suggestion
    <string name="cant_modify_if_refreshing">Please wait until the upload finishes</string>
```

**Comment by teolemon** [comment]
image

**Comment by teolemon** [comment]
clap clap clap 👍

**Pull request merged**